Src/WindowsServer/WindowsServer.Shared.Tests/AzureInstanceMetadataTests.cs

```diff
@@ -0,0 +1,53 @@
+namespace Microsoft.ApplicationInsights.WindowsServer
+{
+    using Microsoft.ApplicationInsights.WindowsServer.Implementation;
+    using Microsoft.ApplicationInsights.WindowsServer.Mock;
+    using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+    using Assert = Xunit.Assert;
+
+    [TestClass]
+    public class AzureInstanceMetadataTests
+    {
+        [TestMethod]
+        public void GetAzureInstanceMetadataFieldsAsExpected()
+        {
+            HeartbeatProviderMock hbeatMock = new HeartbeatProviderMock();
+            AzureInstanceMetadataRequestMock azureInstanceRequestorMock = new AzureInstanceMetadataRequestMock();
+            AzureHeartbeatProperties azureIMSFields = new AzureHeartbeatProperties(azureInstanceRequestorMock, true);
+            int counter = 1;
+            foreach (string field in azureIMSFields.DefaultFields)
+            {
+                azureInstanceRequestorMock.ComputeFields.Add(field, $"testValue{counter++}");
```

Comment:
Did you plan to validate the value of the counter as part of the test later on?

Reply:
No, this was a completely unnecessary addition. I've removed it.